src/Strategy/Elixir/MixTree.hs

```diff
@@ -66,13 +67,14 @@ import Text.Megaparsec (
  )
 import Text.Megaparsec.Char (alphaNumChar, char, eol, newline, string)
 import Text.Megaparsec.Char.Lexer qualified as Lexer
-import Types (GraphBreadth (..))
+import Types (DependencyResults (..), GraphBreadth (..))
 
 missingDepVersionsMsg :: Text
 missingDepVersionsMsg = "Some of dependencies versions were not resolved from `mix deps` and `mix deps.tree`. Has `mix deps.get` and `mix compile` been executed?"
 
-analyze :: (Has Exec sig m, Has Diagnostics sig m, Has Logger sig m) => Path Abs Dir -> m (Graphing Dependency, GraphBreadth)
-analyze dir = do
+analyze :: (Has Exec sig m, Has Diagnostics sig m, Has Logger sig m) => MixProject -> m (DependencyResults)
```

Comment:
[nit] there are lots of redundant parentheses in this PR, e.g. here with `(DependencyResults)`